projects/canopy/src/lib/side-nav/side-nav-bar-item/side-nav-bar-item.component.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  HostBinding,
+  Input,
+  Output,
+  ViewEncapsulation,
+} from '@angular/core';
+
+@Component({
+  selector: 'lg-side-nav-bar-item',
+  templateUrl: './side-nav-bar-item.component.html',
+  styleUrls: ['./side-nav-bar-item.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LgSideNavBarItemComponent {
+  @HostBinding('class.lg-side-nav-bar-item') class = true;
+
+  @Input() isActive = false;
+  @HostBinding('class.lg-side-nav-bar-item--selected') get isActiveClass() {
+    return this.isActive;
+  }
+
+  @Output() clicked: EventEmitter<void> = new EventEmitter();
+
+  onClick() {
```

Comment:
Is it necessary to emit a click event, given you will be implementing the router link in the consumer?

Reply:
Oh I see, you are using it to focus on the content in the parent component

Reply:
Yeah, I use this event to set the focus via the parent component.